lighthouse-core/gather/driver.js

```diff
@@ -143,17 +143,33 @@ class Driver {
         (_ => reject(new Error('The asynchronous expression exceeded the allotted time of 60s'))),
         60000
       );
+
       this.sendCommand('Runtime.evaluate', {
-        expression: asyncExpression,
+        expression: `(function wrapInNativePromise() {
```

Comment:
wanna add a comment to document this? we're definitely gonna be headscratching if we need to touch this code again. :)